src/Perlin.tsx

```diff
@@ -104,15 +106,36 @@ class Perlin {
     private keys: nacl.SignKeyPair;
     private transactionDebounceIntv: number = 2000;
     private peerPollIntv: number = 5000;
+    private interval: any;
 
     private constructor() {
+        const secret = storage.getSecretKey();
+
+        if (secret) {
+            this.setSecretKey(secret);
+        }
+    }
+
+    @action.bound
+    public async setSecretKey(hexString: string): Promise<any> {
         this.keys = nacl.sign.keyPair.fromSecretKey(
-            Buffer.from(
-                "87a6813c3b4cf534b6ae82db9b1409fa7dbd5c13dba5858970b56084c4a930eb400056ee68a7cc2695222df05ea76875bc27ec6e61e8e62317c336157019c405",
-                "hex"
-            )
+            Buffer.from(hexString, "hex")
         );
-        this.init().catch(err => console.error(err));
+        try {
+            await this.init();
+            this.isLogged = true;
```

Comment:
Rather than keeping track of `isLogged`, why not just check if a private key is stored?

Reply:
hmm, I'll check if I can remove it. At the beginning I just want it separated in case of we want to encrypt a private key with specific algorithm.